dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java

```diff
@@ -711,52 +711,53 @@ public String createLocationQueryForAdministrableItems(Context context)
 
             List<Collection> allCollections = new ArrayList<>();
             
-            for( ResourcePolicy rp: collectionsPolicies){
-                Collection collection = ContentServiceFactory.getInstance().getCollectionService()
-                        .find(context, rp.getdSpaceObject().getID());
-                allCollections.add(collection);
-            }
-
-            if (CollectionUtils.isNotEmpty(communitiesPolicies) || CollectionUtils.isNotEmpty(allCollections)) 
-            {
-                locationQuery.append("location:( ");
-
-                for (int i = 0; i< communitiesPolicies.size(); i++) 
-                {
-                    ResourcePolicy rp = communitiesPolicies.get(i);
-                    Community community = ContentServiceFactory.getInstance().getCommunityService()
-                            .find(context, rp.getdSpaceObject().getID());
-                
-                    locationQuery.append("m").append(community.getID());
-
-                    if (i != (communitiesPolicies.size() - 1)) {
-                        locationQuery.append(" OR ");
-                    }
-                    allCollections.addAll(ContentServiceFactory.getInstance().getCommunityService()
-                            .getAllCollections(context, community));
-                }
-
-                Iterator<Collection> collIter = allCollections.iterator();
-
-                if (communitiesPolicies.size() > 0 && allCollections.size() > 0) {
-                    locationQuery.append(" OR ");
-                }
-
-                while (collIter.hasNext()) {
-                    locationQuery.append("l").append(collIter.next().getID());
-
-                    if (collIter.hasNext()) {
-                        locationQuery.append(" OR ");
-                    }
-                }
-                locationQuery.append(")");
+            if(communitiesPolicies.size() != 0 || collectionsPolicies.size() != 0){
+            	
+            	locationQuery.append("location:( ");
+            	
+            	locationQuery.append(this.getLocationOfEachResourcePolicy(communitiesPolicies, context));
+            	
+            	locationQuery.append(this.getLocationOfEachResourcePolicy(collectionsPolicies, context));
```

Comment:
Aside from the issue reported above, when I test with this patch, the same number of objects are returned when only the root location is included in the search, so I think this PR should be merged once the bug is fixed.